locallib.php

```diff
@@ -106,27 +106,89 @@ function hvp_get_core_assets($context) {
     $settings['loadedJs'] = array();
     $settings['loadedCss'] = array();
 
-    // Make sure files are reloaded for each plugin update.
-    $cachebuster = \hvp_get_cache_buster();
-
     // Use relative URL to support both http and https.
     $liburl = \mod_hvp\view_assets::getsiteroot() . '/mod/hvp/library/';
     $relpath = '/' . preg_replace('/^[^:]+:\/\/[^\/]+\//', '', $liburl);
 
     // Add core stylesheets.
     foreach (\H5PCore::$styles as $style) {
-        $settings['core']['styles'][] = $relpath . $style . $cachebuster;
-        $PAGE->requires->css(new moodle_url($liburl . $style . $cachebuster));
+        $url = generate_css_url('library/' . $style);
+        $settings['core']['styles'][] = $url->out(false);
+        $PAGE->requires->css($url);
     }
     // Add core JavaScript.
     foreach (\H5PCore::$scripts as $script) {
-        $settings['core']['scripts'][] = $relpath . $script . $cachebuster;
-        $PAGE->requires->js(new moodle_url($liburl . $script . $cachebuster), true);
+        $scriptpath = $relpath . $script;
+        $settings['core']['scripts'][] = generate_js_url($scriptpath)->out(false);
+        $PAGE->requires->js($scriptpath, true);
     }
 
     return $settings;
 }
 
+/**
+ * Determine the correct JS Revision to use for this load.
+ * Copied from outputrequirementslib because that function is protected.
+ *
+ * @return int the jsrev to use.
+ */
+function get_jsrev(): int {
```

Comment:
can't this just be $jsrev = $PAGE->requires->get_jsrev();  ?

Reply:
get_jsrev is a protected function. It cannot be accessed this way.

Reply:
Are you working on a clean master moodle? It is public in core and has been since 3.9+ https://tracker.moodle.org/browse/MDL-67063

Doesn't cover totara though :/

Can you make a note about this in a comment

Reply:
I am in the habit of using older versions to ensure backward compatibility. I will change to use the master branch. What I would really like is for js_fix_url() to be public.

For comment, do you mean in the README?

Reply:
No I mean comment right here in the code saying where it was copied from and why. Even say the conditions under which it could be removed, eg 'This is needed for Totara support, and Moodle prior to 3.9'